### PR TITLE
feat(commitlint): add husky commit-msg with commitlint configuration

### LINKED ISSUE #118: Enforce semantic commit messages
Using Husky and commitlint, commit messages should be enforced using a semantic style:

`<type>(<scope>): <subject>, not this: (<scope>) <type>: <subject>`

### PR DESCRIPTION
**Pull request by Jayanth-Parthsarathy**
# Title
Fixes #118 .
Added `commitlint` and `husky` for linting commit messages using the semantics:

`<type>(<scope>): <subject>`

## Screenshots

image

### CODE REVIEW

File: package.json
Comment (hwgilbert16) [inline package.json:1]: Move the new commit linting dependencies to devDependencies

**Review by hwgilbert16 — CHANGES_REQUESTED** [review]
Looks good, but just move the new dependencies you added to devDependencies

File: package.json
Comment (Jayanth-Parthsarathy) [reply package.json:1]: > Move the new commit linting dependencies to devDependencies

Hey @hwgilbert16. I think the commit lint dependencies are already under `devDependencies`. Please feel free to correct me if I am wrong.

Comment (hwgilbert16) [reply package.json:1]: Ah, apologies. The web diff viewer made it seem as though the new dependencies were not under the correct category.

**Review by hwgilbert16 — APPROVED** [review]

**Pull request merged**